### PR TITLE
EPMRPP-90025 || Bug fix sidebar

### PR DESCRIPTION
**Pull request by BlazarQSO**

### CODE REVIEW

File: app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx [hunk 1]
```diff
@@ -42,12 +44,35 @@ export const OrganizationsPopover = ({ closePopover, closeNavbar }) => {
...
+
+  const [scrollTop, setScrollTop] = useState(0);
 
   const onClose = () => {
     closeNavbar();
     closePopover();
   };
 
+  useLayoutEffect(() => {
+    availableProjects.forEach(({ organizationSlug, projects }, organizationItems) => {
```
Comment (Vadim73i) [inline app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx:57]:
```
availableProjects
.filter(({ organizationSlug }) => currentOrganization === organizationSlug)
.forEach(({ projects }, organizationItems) => {
    projects.forEach(..........
```

File: app/src/componentLibrary/sidebar/navbar/navbar.jsx [hunk 2]
```diff
@@ -43,11 +48,15 @@ export const Navbar = ({
       {topSidebarControlItems.map(({ sidebarBlockItem, key, onClick }) => (
         <SidebarButton
           key={key}
-          className={cx('navbar-btn')}
+          className={cx('navbar-btn', getClassName(key, true))}
```
Comment (AmsterGet) [inline app/src/componentLibrary/sidebar/navbar/navbar.jsx:51]: Please rewrite this and avoid mystery boolean arguments.

File: app/src/componentLibrary/sidebar/navbar/navbar.jsx [hunk 3]
```diff
@@ -80,6 +93,11 @@ Navbar.propTypes = {
   mainControlBlock: PropTypes.element,
   footerControlBlock: PropTypes.element,
   setIsOpenPopover: PropTypes.func,
+  getClassName: PropTypes.object.isRequired,
+  onEnterButton: PropTypes.func.isRequired,
+  onLeaveButton: PropTypes.func.isRequired,
+  onButtonDown: PropTypes.func.isRequired,
+  onButtonUp: PropTypes.func.isRequired,
```
Comment (AmsterGet) [inline app/src/componentLibrary/sidebar/navbar/navbar.jsx:100]: Please align naming to be consistent with event handlers names.

File: app/src/componentLibrary/sidebar/sidebarButton/sidebarButton.jsx [hunk 4]
```diff
@@ -20,8 +20,24 @@ import styles from './sidebarButton.scss';
...
+  children,
+}) => (
+  <button
+    className={cx('sidebar-button', className)}
+    onClick={onClick}
+    tabIndex={-1}
+    onMouseEnter={onMouseEnter}
+    onMouseLeave={onMouseLeave}
+    onMouseDown={onMouseDown}
+    onMouseUp={onButtonUp}
```
Comment (AmsterGet) [inline app/src/componentLibrary/sidebar/sidebarButton/sidebarButton.jsx:39]: Same here about naming.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/componentLibrary/sidebar/sidebar.jsx [hunk 5]
```diff
@@ -65,6 +71,42 @@ export const Sidebar = ({
...
+  const onButtonUp = () => {
+    setActionButtonKey(null);
+    setActionButtonType(null);
+  };
+
+  const getClassName = (key, isNavbar = false) => {
+    const isAction = actionButtonKey === key;
+
+    return cx({
+      'sidebar-btn': !isNavbar,
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/componentLibrary/sidebar/sidebar.jsx:104]: Removed this property

File: app/src/componentLibrary/sidebar/sidebar.jsx [hunk 6]
```diff
@@ -65,6 +71,42 @@ export const Sidebar = ({
...
+    onCloseNavbar();
+  };
+
+  const onEnterButton = (key) => {
+    setActionButtonKey(key);
+    setActionButtonType(HOVER);
+  };
+
+  const onLeaveButton = () => {
+    setActionButtonKey(null);
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/componentLibrary/sidebar/sidebar.jsx:86]: Done.

File: app/src/controllers/user/selectors.js [hunk 7]
```diff
@@ -50,7 +50,7 @@ export const availableProjectsSelector = createSelector(
     );
 
     return Object.keys(assignedOrganizations)
-      .sort()
+      .sort((a, b) => (a.toLowerCase() > b.toLowerCase() ? 1 : -1))
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/controllers/user/selectors.js:53]: Done.

File: app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx (hunk 1 above)
Comment (BlazarQSO) [reply app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx:57]: Removed

File: app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx [hunk 8]
```diff
@@ -42,12 +44,35 @@ export const OrganizationsPopover = ({ closePopover, closeNavbar }) => {
...
+  const maxHeightPopover = window.innerHeight - MARGIN_TOP_AND_MARGIN_BOTTOM;
+
+  const [scrollTop, setScrollTop] = useState(0);
 
   const onClose = () => {
     closeNavbar();
     closePopover();
   };
 
+  useLayoutEffect(() => {
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx:56]: Removed

File: app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx [hunk 9]
```diff
@@ -70,9 +95,11 @@ export const OrganizationsPopover = ({ closePopover, closeNavbar }) => {
       <ScrollWrapper
         autoHide
         autoHeight
-        autoHeightMax={window.innerHeight - MARGIN_TOP_AND_MARGIN_BOTTOM}
+        autoHeightMax={maxHeightPopover}
         hideTracksWhenNotNeeded
         className={cx('scroll-wrapper')}
+        defaultScrollTop={scrollTop}
+        key={scrollTop}
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/layouts/appLayout/projectSidebar/organizationsControl/organizationsPopover/organizationsPopover.jsx:102]: Removed

File: app/src/componentLibrary/sidebar/navbar/navbar.jsx (hunk 2 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/sidebar/navbar/navbar.jsx:51]: Done.

File: app/src/componentLibrary/sidebar/navbar/navbar.jsx (hunk 3 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/sidebar/navbar/navbar.jsx:100]: Done.

File: app/src/componentLibrary/sidebar/sidebarButton/sidebarButton.jsx (hunk 4 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/sidebar/sidebarButton/sidebarButton.jsx:39]: Done.

**Review by AmsterGet — APPROVED** [review]

**Review by Vadim73i — APPROVED** [review]

**Pull request merged**